x-pack/plugins/maps/public/classes/sources/mvt_single_layer_vector_source/mvt_single_layer_source_settings.tsx

```diff
@@ -0,0 +1,218 @@
+import { EuiFieldText, EuiFormRow } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import _ from 'lodash';
+import { MAX_ZOOM, MIN_ZOOM } from '../../../../common/constants';
+import { ValidatedDualRange, Value } from '../../../../../../../src/plugins/kibana_react/public';
+import { MVTFieldConfigEditor } from './mvt_field_config_editor';
+import { MVTFieldDescriptor } from '../../../../common/descriptor_types';
+
+export type MVTSettings = {
+  layerName: string;
+  fields: MVTFieldDescriptor[];
+  minSourceZoom: number;
+  maxSourceZoom: number;
+};
+
+export interface State {
+  currentLayerName: string;
+  currentMinSourceZoom: number;
+  currentMaxSourceZoom: number;
+  currentFields: MVTFieldDescriptor[];
+  previousLayerName: string;
+  previousMinSourceZoom: number;
+  previousMaxSourceZoom: number;
+  previousFields: MVTFieldDescriptor[];
+}
+
+export interface Props {
+  handleChange: (args: MVTSettings) => void;
+  layerName: string;
+  fields: MVTFieldDescriptor[];
+  minSourceZoom: number;
+  maxSourceZoom: number;
+  includeFields: boolean;
+}
+
+export class MVTSingleLayerSourceSettings extends Component<Props, State> {
+  state = {
+    currentLayerName: '',
+    currentMinSourceZoom: MIN_ZOOM,
+    currentMaxSourceZoom: MAX_ZOOM,
+    currentFields: [],
+    previousLayerName: '',
+    previousMinSourceZoom: MIN_ZOOM,
+    previousMaxSourceZoom: MAX_ZOOM,
+    previousFields: [],
+  };
+
+  static getDerivedStateFromProps(nextProps: Props, prevState: State) {
+    const newSettings = {
+      layerName: nextProps.layerName,
+      fields: nextProps.fields,
+      minSourceZoom: nextProps.minSourceZoom,
+      maxSourceZoom: nextProps.maxSourceZoom,
+    };
+
+    const previous = prevState
+      ? {
+          layerName: prevState.previousLayerName,
+          fields: prevState.previousFields,
+          minSourceZoom: prevState.previousMinSourceZoom,
+          maxSourceZoom: prevState.previousMaxSourceZoom,
+        }
+      : null;
+
+    if (_.isEqual(previous, newSettings)) {
+      return null;
+    }
+
+    const clonedFields = _.cloneDeep(nextProps.fields);
+    return {
+      currentLayerName: nextProps.layerName,
+      currentMinSourceZoom: nextProps.minSourceZoom,
+      currentMaxSourceZoom: nextProps.maxSourceZoom,
+      currentFields: clonedFields,
+      previousLayerName: nextProps.layerName,
+      previousMinSourceZoom: nextProps.minSourceZoom,
+      previousMaxSourceZoom: nextProps.maxSourceZoom,
+      previousFields: clonedFields,
+    };
+  }
+
+  _handleChange = _.debounce(() => {
+    this.props.handleChange({
+      layerName: this.state.currentLayerName,
+      minSourceZoom: this.state.currentMinSourceZoom,
+      maxSourceZoom: this.state.currentMaxSourceZoom,
+      fields: this.state.currentFields,
+    });
+  }, 200);
+
+  _handleLayerNameInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const layerName = e.target.value;
+    if (layerName === this.state.currentLayerName) {
```

Comment:
Checks like these are not needed in your onChange handlers. This check can be removed.